### ISSUE TITLE
NVDA menu and NVDA settings categories cannot be accessed when certain NVDA message dialogs are displayed

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#14768]
### Steps to reproduce:
1. Open either the info dialog in NVDA menu / help, or the COM registration fixing tool in NVDA menu / tools
2. Press nvda+n to open the NVDA menu

### Actual behavior:
NVDA menu does not open.

### Expected behavior:
Either NVDA menu should be displayed,, or NVDA should report "this action cannot be performed while a dialog asks for a confirmation" like it does for dialog opening commands such as nvda+ctrl+g, d, s etc.

### NVDA logs, crash dumps and other attachments:
n/a

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2023.1

#### Windows version:
Windows 11 22 H2

#### Does the issue still occur after restarting your computer?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no

#### If NVDA add-ons are disabled, is your problem still occurring?
yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by Adriani90** [comment]
Actually doing the vice versa seems to work,
1. Open a settings dialog
2. Open the NVDA menu
3. Open a confirmation dialog like info dialog or cOM registration fixing tool.


Why does one way work but the other way round fails?

**Comment by CyrilleB79** [comment]
> Actually doing the vice versa seems to work,
> 
> 1. Open a settings dialog
> 2. Open the NVDA menu
> 3. Open a confirmation dialog like info dialog or cOM registration fixing tool.
> 
> Why does one way work but the other way round fails?

I do not understand your vice versa scenario. Could you ensure that you have listed all the steps, and overall indicated expected and actual results? Thanks.

**Comment by Adriani90** [comment]
@CyrilleB79 thanks for the comment, I've edited the title and description to refelct my comment regarding the other way raound.

**Comment by jcsteh** [comment]
Certain dialogs such as the About dialog are "modal" dialogs. They are intended to be opened and dismissed without doing anything else in between. Note that this is not unique to NVDA. Various message dialogs, file open/save dialogs, etc. in other applications are also modal and you can't do anything else with the application while these are open.

While I guess having an explicit message about this could be useful for some users, I don't think it offers a huge usability benefit:

1. I can't think of many common use cases where opening the NVDA menu while one of these dialogs open is particularly useful.
2. Users are going to realise fairly quickly that these scenarios don't work. A message will make that explicit, but it won't really tell them anything they don't already know.
3. As above, this pattern is established in other applications.

Given the above reasons, I'm prioritising this as p4.
